posts/Stef/2019-11-18-hibernate-orm-with-panache-in-quarkus.adoc

```diff
@@ -0,0 +1,465 @@
+- Keep entity model refactorying within a single file
+- Do not require injection to manipulate them (does not break the editing flow)
+- Have great discoverability: just type the entity type and complete to get all methods
+
+Let's review our new entity class:
+
+[source%nowrap,java]
+----
+@Entity
+public class Person extends PanacheEntity {
+
+    public String firstName;
+    public String lastName;
+    public Date birth;
+    
+    public static List<Person> findByName(String lastName){
+        return find("lastName", lastName).list();
+    }
+    
+    public static List<Person> findBornAfter(Date date){
+        return find("birth > :date", Parameters.with("birth", date)).list();
+    }
+}
+----
+
+And this is now our REST endpoint:
+
+[source%nowrap,java]
+----
+@Path("/")
+public class PersonEndpoint {
+    
+    @GET
+    @Path("people")
+    public List<Person> all(){
+        return Person.findAll().list();
+    }
+
+    @GET
+    @Path("people/by-name")
+    public List<Person> findByName(@PathParam String name){
+        return Person.findByName(name);
+    }
+
+    @GET
+    @Path("people/born-after")
+    public List<Person> findBornAfter(@PathParam Date date){
+        return Person.findBornAfter(date);
+    }
+
+    @GET
+    @Path("person/{id}")
+    public Person findById(@PathParam Long id){
+        Person p = Person.findById(id);
+        if(p == null)
+            throw new WebApplicationException(Status.NOT_FOUND);
+        return p;
+    }
+
+    @PUT
+    @Path("person/{id}")
+    public void updatePerson(@PathParam Long id, Person newPerson){
+        Person p = Person.findById(id);
+        if(p == null)
+            throw new WebApplicationException(Status.NOT_FOUND);
+        p.birth = newPerson.birth;
+        p.firstName = newPerson.firstName;
+        p.lastName = newPerson.lastName;
+    }
+
+    @DELETE
+    @Path("person/{id}")
+    public void deletePerson(@PathParam Long id){
+        Person p = Person.findById(id);
+        if(p == null)
+            throw new WebApplicationException(Status.NOT_FOUND);
+        p.delete();
+    }
+    
+    @POST
+    @Path("people")
+    public Response newPerson(@Context UriInfo uriInfo, Person newPerson){
+        Person p = new Person();
+        p.birth = newPerson.birth;
+        p.firstName = newPerson.firstName;
+        p.lastName = newPerson.lastName;
+        Person.persist(p);
+        
+        URI uri = uriInfo.getAbsolutePathBuilder()
+                .path(PersonEndpoint.class)
+                .path(PersonEndpoint.class, "findById")
+                .build(p.id);
+        return Response.created(uri).build();
+    }
+}
+----
+
+How does this look?
+
+This is the Hibernate ORM we've come to love with its solid core and tons of features, with just enough varnish to make it simpler
+to write your data layer. That's what we call “with Panache”.
+
+Go and start building your https://code.quarkus.io/[first Quarkus application today] with
```

Comment:
It's a nice post, but I feel like after getting people excited the end is a bit rushed :)

Maybe expand a little bit on how to get started? You're just tossing the linke there; maybe add a suggestion to pick the Panache component and a database component, or introduce people to the quickstart.

Reply:
good points.

Reply:
I don't think we have a quickstart for it, but I've added some text about the code generator.

Reply:
https://github.com/quarkusio/quarkus-quickstarts/tree/master/hibernate-orm-panache-quickstart

Reply:
dafuk. Now you're telling me I should update https://quarkus.io/guides/hibernate-orm-panache to mention this quickstart. I thought I'd gotten lucky by not having to debug some Ruby gems but only dodged that bullet to catch others :(

Reply:
Even worse: I now need to learn how to use a quickstart ;)

Reply:
Oh, I know why it's not listed there: I copied https://quarkus.io/guides/hibernate-orm which does not mention the quickstart! HAH

Reply:
LOL. As you prefer, I didn't mean it as pressure to update the guide. Just suggesting you link to it from your blog.. assuming the quickstart is fine?

I'm worried now since you didn't know about that one, wondering if we should check it out.